### DISCUSSION TITLE
Any plan to follow ERC20 or the Hyperledger Token Taxonomy Framework?

### DISCUSSION THREAD

**Discussion by yeasy** [hyperledger-labs/fabric-token-sdk#38] [Ideas]
Started in issue #32 , this may be a point worthy of discussions.

**Comment by adecaro** [comment]
@yeasy, thanks for starting this discussion. 

- Regarding ERC20, that standard allows smart contracts, in the same Ethereum network, to interoperate. So, a smart contract that delivers a certain functionality can use an ERC20 smart contract for the payments. In Fabric, these kinds of interactions are usually more complex to achieve given the different endorsement policies chaincodes might have.  However thanks to the Fabric Smart Client, we can achieve these kinds of interactions more easily. Please, have a look at this [`DvP example`](https://github.com/hyperledger-labs/fabric-token-sdk/tree/main/integration/token/dvp) (The documentation will be ready ASAP).  
On the other hand, if we put ourselves in the shoes of a client application, the Token SDK offers wallets to control the tokens and compute balances. 
- Regarding the Token Taxonomy, that's very interesting. One of the goals of the Token SDK is to deliver token related operation with privacy. Not all operations described by the Token Taxonomy are easy to deploy with privacy (using ZK, for example). Therefore, we have to pay attention. Definitely, it would be interesting to map what currently the Token SDK delivers to that framework and to see which other operations we can support in addition.

What's your take on this?

**Reply by yeasy** [reply]
Thanks for the comments!
After reading the code, i feel that the token sdk heavily relies on the smart client, which offer quite different operation models compared with legacy clients, e.g., it can eliminate peer's endorsement at some case.

And token sdk claims to be good in the interoperability, hence we should seriously consider to follow at least one "standard" from the beginning. If TTF lacks supports on privacy, maybe we can propose the requirement to update it.

**Reply by adecaro** [reply]
It makes sense to me. Do you have a proposal in mind, already?

**Reply by yeasy** [reply]
Not yet. If you think it makes sense, we can create one issue to track the progress first.
